### ISSUE TITLE
Client waits 10s then reports "Failed to free 1 iomgr objects before shutdown deadline"

### ISSUE THREAD

**Issue by kkm000** [grpc/grpc#2504]
Running the C# Greeter example from [grpc-common](https://github.com/grpc/grpc-common/tree/master/csharp) under VS2013 update 4, Windows 7 64-bit. Server started, running the client:

```
GRPC initialized.
Greeting: Hello you
Waiting for GPRC threads to finish.
Completion queue has shutdown successfully, thread grpc 1 exiting.
Completion queue has shutdown successfully, thread grpc 2 exiting.
Completion queue has shutdown successfully, thread grpc 3 exiting.
Completion queue has shutdown successfully, thread grpc 0 exiting.
D150718 170901 0 c:\users\jtattermusch\github\grpc\src\core\iomgr\iomgr.c:119: Waiting for 1 iomgr objects to be destroyed and executing final callbacks
D150718 170901 0 c:\users\jtattermusch\github\grpc\src\core\iomgr\iomgr.c:126: Waiting for 1 iomgr objects to be destroyed
```
Then the client hangs for about 10 seconds, and more messages printed:
```
D150718 170911 0 c:\users\jtattermusch\github\grpc\src\core\iomgr\iomgr.c:156: Failed to free 1 iomgr objects before shutdown deadline: memory leaks are likely
D150718 170911 0 c:\users\jtattermusch\github\grpc\src\core\iomgr\iomgr.c:161: LEAKED OBJECT: client 003CC388
D150718 170911 3897240 c:\users\jtattermusch\github\grpc\src\core\iomgr\iocp_windows.c:130: iocp_loop is done
GRPC shutdown.
```
At the same moment, server prints the following error message
```
E150718 170911 8549904 c:\users\jtattermusch\github\grpc\src\core\iomgr\tcp_windows.c:152: ReadFile overlapped error: An existing connection was forcibly closed by the remote host.
```
NuGet packages:
```xml
<packages>
  <package id="Google.ProtocolBuffers" version="2.4.1.555" targetFramework="net45" />
  <package id="Grpc" version="0.6.0" targetFramework="net45" />
  <package id="Grpc.Core" version="0.6.0" targetFramework="net45" />
  <package id="grpc.dependencies.openssl.redist" version="1.0.2.2" targetFramework="net45" />
  <package id="grpc.dependencies.zlib.redist" version="1.2.8.9" targetFramework="net45" />
  <package id="grpc.native.csharp_ext" version="0.10.0" targetFramework="net45" />
  <package id="Ix-Async" version="1.2.3" targetFramework="net45" />
  <package id="System.Collections.Immutable" version="1.1.36" targetFramework="net45" />
</packages>
```

**Comment by jtattermusch** [comment]
Hi, thanks for reporting this! Nevertheless, this is a known bug (see #1650) that has already been fixed (#2416). As I can see this bug can be pretty annoying, we will consider doing a 0.6.1 release of gRPC C# that would have this fixed (0.6 was released before fixing the bug).

**Comment by kkm000** [comment]
Sorry, I did a few queries, but could not find the other tickets.
